services/spar/test-integration/Util.hs

```diff
@@ -452,37 +453,59 @@ createTestIdPFrom metadata mgr brig galley spar = do
     (uid, tid,) <$> callIdpCreate spar (Just uid) metadata
 
 
+getCookie :: KnownSymbol name => proxy name -> ResponseLBS -> Either String (SAML.SimpleSetCookie name)
+getCookie proxy rsp = do
+  web :: Web.SetCookie
+    <- Web.parseSetCookie <$> maybe (Left "no set-cookie header") Right
+                                    (lookup "set-cookie" (responseHeaders rsp))
+  if Web.setCookieName web == SAML.cookieName proxy
+    then Right $ SimpleSetCookie web
+    else Left $ "bad cookie name.  (found, expected) == " <> show (Web.setCookieName web, SAML.cookieName proxy)
+
+-- | In 'setResponseCookie' we set no expiration date iff cookie is persistent.  So here we test for
+-- expiration date.  Easier than parsing and inspecting the cookie value.
+hasPersistentCookieHeader :: ResponseLBS -> Either String ()
+hasPersistentCookieHeader rsp = do
+  cky <- getCookie (Proxy @"zuid") rsp
+  unless (isNothing . Web.setCookieExpires $ fromSimpleSetCookie cky) $
+    Left $ "expiration date shoult be empty: " <> show cky
```

Comment:
This test tests exactly the opposite of what it should be testing here:

When you log in **without** the `persist=true` query parameter, or with persist=false, you get a `session cookie`, which means it has no expiration date set, and will expire when you close the browser (and on the backend has a validity of max 1 week):
```
POST /login?persist=false HTTP/1.1

Set-Cookie: zuid=(redacted); Path=/access; Domain=zinfra.io; HttpOnly; Secure
```

When you log in **with** `persist=true`, you get a persistent cookie, which means it has *some* expiration date (in prod that's 56 days, in integration test that's a few seconds or minutes) and can be renewed with `POST /access`:
```
POST /login?persist=true HTTP/1.1

Set-Cookie: zuid=(redacted); Path=/access; Expires=Thu, 10-Jan-2019 10:43:28 GMT; Domain=zinfra.io; HttpOnly; Secure
```

Can you adapt your test code here and the comments to reflect this?